gravitee-am-gateway/gravitee-am-gateway-handler/gravitee-am-gateway-handler-oidc/src/main/java/io/gravitee/am/gateway/handler/oauth2/service/response/HybridResponse.java

```diff
@@ -70,6 +68,7 @@ public void setCode(String code) {
     @Override
     public String buildRedirectUri() {
         UriBuilder uriBuilder = UriBuilder.fromURIString(getRedirectUri());
+        // Hybrid response requires Fragment
```

Comment:
Not sure if we should also consider the response mode here ... Difficult to read in the spec if the response_mode override the default response mode of the Flow. 

WDYT ?

Reply:
same as for id_token,

For all of them we have this sentence : `The default Response Mode for this Response Type is the fragment encoding and the query encoding MUST NOT be used`


`
This section defines combinations of the values code, token, and id_token, which are each individually registered Response Types.

code token
When supplied as the value for the response_type parameter, a successful response MUST include an Access Token, an Access Token Type, and an Authorization Code. The default Response Mode for this Response Type is the fragment encoding and the query encoding MUST NOT be used. Both successful and error responses SHOULD be returned using the supplied Response Mode, or if none is supplied, using the default Response Mode.

code id_token
When supplied as the value for the response_type parameter, a successful response MUST include both an Authorization Code and an id_token. The default Response Mode for this Response Type is the fragment encoding and the query encoding MUST NOT be used. Both successful and error responses SHOULD be returned using the supplied Response Mode, or if none is supplied, using the default Response Mode.

id_token token
When supplied as the value for the response_type parameter, a successful response MUST include an Access Token, an Access Token Type, and an id_token. The default Response Mode for this Response Type is the fragment encoding and the query encoding MUST NOT be used. Both successful and error responses SHOULD be returned using the supplied Response Mode, or if none is supplied, using the default Response Mode.

code id_token token
When supplied as the value for the response_type parameter, a successful response MUST include an Authorization Code, an id_token, an Access Token, and an Access Token Type. The default Response Mode for this Response Type is the fragment encoding and the query encoding MUST NOT be used. Both successful and error responses SHOULD be returned using the supplied Response Mode, or if none is supplied, using the default Response Mode.
`

Reply:
I just checking also the JARM rfc, query.jwt is forbidden also if token or id_token are part of the response_type 
https://openid.net/specs/oauth-v2-jarm.html#name-response-mode-queryjwt

Will amend the validation method to check these combinaison if it is not already done.

Reply:
Same thing as above : 

```Both successful and error responses SHOULD be returned using the supplied Response Mode, or if none is supplied, using the default Response Mode.```

Not sure how to read this ... but I still think we should rely on the supplied response mode.

Reply:
OK after digging into the differents specs, here is the conclusion:

If `token` or `id_token` are part of the `response_type`, then the `response_mode` has to be `fragment` (or `fragment.jwt`) for security reason as spotted by https://openid.net/specs/oauth-v2-multiple-response-types-1_0.html#Security.